tron/serialize/runstate/mirror_state_store.py

```diff
@@ -0,0 +1,73 @@
+import copy
+import pickle
+from threading import Thread
+
+
+class MirrorStateStore():
+    def __init__(self, shelve_store, dynamodb_store) -> None:
+        self.shelve_store = shelve_store
+        self.dynamodb_store = dynamodb_store
+
+    def is_valid(self, shelve_kv_pairs: dict, dynamo_kv_pairs: dict) -> bool:
+        """
+        It checks if all keys in DynamoDB are in shelveStateStore and have the same values.
+        It is possible that some keys are in shelveStateStore but is not in DynamoDB when the
```

Comment:
How will you know when you're ready to stop using shelve, then? (when all keys in shelve are in dynamo)

Reply:
I assume unless dynamodb read/write fails, where an exception is thrown and I'll be alerted, all keys in shelve are in dynamodb if they are ever written.